### DISCUSSION TITLE
Support for NDI 6

### DISCUSSION THREAD

**Discussion by ruchit-bhatt** [ggarra13/mrv2#177] [Ideas]
Can we get support for NDI stream ? so that artists can review Premier-pro, Resolve, Avid, etc timeline remotely.
NDI 6 now also supports wide color gamut, PQ and HLG formats.

Links
https://ndi.video/tech/ndi6/
https://www.louper.io/
https://www.drastic.tv/productsmenu-56/networkstreaminglist/flowcaster
https://www.streambox.com/

**Comment by ggarra13** [comment]
Sigh.  I want to retire from programming, but I am forced into it time and again :)

I already found out from their web page you linked that HDR support is a paid subscription service, so you would not be getting wide color gamut but only SDR by default.

My first question for you would be if you want to review the NLE editors' timeline on a LAN (ie. your artists working on a private network) or on the Web (ie. remotely).   Adding support for artists on a LAN should be easier, but adding support for artists (or directors, etc) on the web easily (ie. without network technical know-how on the artist/director side) is not currently supported by NDI and would require you to hire a web service like louper.io, so I would not be able to offer you any benefits.

**Reply by ruchit-bhatt** [reply]
Oh, 🤓
I want LAN based option only,

**Comment by ggarra13** [comment]
I have not forgotten about you.  I wanted to investigate and have a basic implementation in place, before we proceed.  

1) What platforms do you want suppported?
2) The streaming input must be from a source that updates the video on every frame (like the output from playback of a video editor), and not from a video capture or video test signal that sends a frame every now and then.  This is a current limitation of the tlRender library.
3) Are you willing to pay for a custom version of mrv2 with NDI support for you to have a competitive advantage?

**Comment by ggarra13** [comment]
I added support for NDI 5 sources playback in mrv2.  It works relatively well for HD sources.  In the next version 4K sources will also be supported.
I am looking for some feedback how it works for you, as I tested on a local wi-fi intranet which is the most awful LAN you can have.

This is what I tested:

1) VLC playing sources (from the NDI Tools.exe) to all platforms (Windows, macOS and Linux).  I tested with the Creative Commons Big Buck Bunny movie and with Charge.
2) ffmpeg with the libndi_newtek device plugin to local Linux.  I was unable to make it work on the network (probably some network misconfiguration on my part).  I tested it with all my clips.
3) macOS test signal to Windows or Linux.  It shows the first frame, but since it is a test signal it does not play it (currently a limitation of mrv2).

**Comment by ggarra13** [comment]
Resurrecting this old thread.  v1.3.4 has added NDI 6 output (except macOS arm64).  The pre-compiled versions support only 8-bits.  I am waiting on confirmation from Vizr (NDI creators) whether the Advanced SDK (which supports full HDR) can be distributed in a BSD-3 application (it seems it might).

**Comment by ggarra13** [comment]
I have released v1.3.5 with HDR support, as the NDI 6 Advanced SDK is not restrictive for open source projects.
